### PR TITLE
PARQUET-2012 Mark ProtoParquetWriter constructors deprecated

### PR DESCRIPTION
**Pull request by dossett**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET-2012) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-2012

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by dossett** [comment]
cc @belugabehr I know you have at least one other open PR that cleans up parquet-protobuf, just want to make sure this doesn't make things worse.

File: parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java [hunk 1]
```diff
@@ -77,26 +81,24 @@ public ProtoParquetWriter(Path file, Class<? extends Message> protoMessage,
...
 	    return new Builder<T>(file);
 	}
 
 	public static <T> Builder<T> builder(OutputFile file) {
 	    return new Builder<T>(file);
 	}
-	
 	private static <T extends MessageOrBuilder> WriteSupport<T> writeSupport(Class<? extends Message> protoMessage) {
-		return new ProtoWriteSupport<T>(protoMessage);
+		return new ProtoWriteSupport<>(protoMessage);
```
Comment (dossett) [inline parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java:99]: Teeny tiny clean-up to remove a redundant type parameter.

File: parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java [hunk 2]
```diff
@@ -43,6 +43,8 @@
    * @param blockSize            HDFS block size
    * @param pageSize             See parquet write up. Blocks are subdivided into pages for alignment and other purposes.
    * @throws IOException if there is an error while writing
+   *
+   * @deprecated will be removed in 2.0.0.; Used ProtoParquetWriter.Builder instead
```
Comment (gszadovszky) [inline parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java:47]: Please, also use the java annotation `@Deprecated` for all the deprecated methods.

**Review by gszadovszky — CHANGES_REQUESTED** [review]
Only one finding. LGTM, otherwise.

File: parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java (hunk 2 above)
Comment (dossett) [reply parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java]: Got it, thanks for spotting that!

**Review by gszadovszky — APPROVED** [review]
Thanks for fixing this

File: parquet-protobuf/src/test/java/org/apache/parquet/proto/TestUtils.java [hunk 3]
```diff
@@ -198,14 +197,11 @@ private static void checkSameBuilderInstance(MessageOrBuilder[] messages) {
...
-  }
-
-  public static Path writeMessages(Class<? extends Message> cls, MessageOrBuilder... records) throws IOException {
     Path file = someTemporaryFilePath();
+    Class<? extends Message> cls = inferRecordsClass(records);
 
-    ProtoParquetWriter<MessageOrBuilder> writer =
-            new ProtoParquetWriter<MessageOrBuilder>(file, cls);
+    ParquetWriter<MessageOrBuilder> writer =
+      ProtoParquetWriter.<MessageOrBuilder>builder(file).withMessage(cls).build();
```
[the remark this answers is not in the document]
Comment (akverma00) [reply parquet-protobuf/src/test/java/org/apache/parquet/proto/TestUtils.java:204]: @dossett 
The purpose was to create ProtoParquetWriter object. Here the return type is itself changed to the parent class type.
If I want to create the "ProtoParquetWriter" type of object, it is not possible using builder pattern, I have to use the deprecated constructor itself as of now.

Can you explain why was this done?

[the remark this answers is not in the document]
Comment (dossett) [reply parquet-protobuf/src/test/java/org/apache/parquet/proto/TestUtils.java:204]: @akverma00 The change was to recognize that using the deprecated constructor, directly or indirectly, was deprecated.  I think what you had to do, using the deprecated constructor, was appropriate and now the deprecation is very clear. I had to do something similar after this change on a project I work on.

**Pull request merged**